shpc/client/__init__.py

```diff
@@ -446,6 +446,13 @@ def get_parser():
         default=None,
         dest="filter_string",
     )
+    show.add_argument(
+        "-l",
+        "--limit",
+        help="set a limit for the number to show",
+        default=None,
+        type=int,
```

Comment:
i guess you may want to enforce its only a positive number and not just any number to avoid passing value like `--limit 0` or negative value

Reply:
This actually isn’t needed because if the user sets a hairbrained limit like that, the loop should just quit immediately.

Reply:
i see

Reply:
LGTM!